src/mappings/entities/mapIdentities.ts

```diff
@@ -361,3 +371,39 @@ const handleSecondaryKeysAdded = async (
 
   await Promise.all(promises);
 };
+
+const handlePrimaryKeyUpdated = async (
+  blockId: string,
+  eventId,
+  params: Codec[]
+): Promise<void> => {
+  const [rawDid, , newKey] = params;
+  const address = getTextValue(newKey);
+
+  const identity = await getIdentity(getTextValue(rawDid));
+
+  Object.assign(identity, {
+    primaryAccount: address,
+    updatedBlockId: blockId,
+    eventId,
+  });
+
+  await identity.save();
+};
+
+const handleSecondaryKeyLeftIdentity = async (params: Codec[]): Promise<void> => {
+  const [, rawAccount] = params;
+
+  const account = JSON.parse(rawAccount.toString()) as MeshAccount;
+
+  let address;
+  if (typeof account === 'string') {
+    // for chain version >= 5.0.0
+    address = account;
+  } else {
+    // for chain version < 5.0.0
+    ({ account: address } = account);
+  }
+
+  await Promise.all([[Account.remove(address), Permissions.remove(address)]]);
```

Comment:
In this scenario, I would only remove the mapping of DID from Account instead of removing the entire account.

Reply:
What do you think @polymath-eric ?